analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -660,12 +660,12 @@ private ProgramState VisitVariableDesignation(VariableDesignationSyntaxWrapper v
                 var variableSymbol = SemanticModel.GetDeclaredSymbol(singleVariableDesignation);
                 var newSymbolicValue = SymbolicValue.Create();
                 newProgramState = SetNewSymbolicValueIfTracked(variableSymbol, newSymbolicValue, newProgramState);
+                // When the pattern is "x is Type t" we know that "t != null", hence (SV != null)
+                newProgramState = newProgramState.SetConstraint(newSymbolicValue, ObjectConstraint.NotNull);
 
                 if (singleVariable)
```

Comment:
this is really confusing:
- I see `if (SingleVariableDesignationSyntaxWrapper.IsInstance(variableDesignation))`
AND
- `if (singleVariable)`

Uhm... could you rename `singleVariable` to something better? From what I understand, this makes the difference between `s1` when inside `x is string s1` vs `x is (string s1, string s2)` - however when it reaches line 668, both are `SingleVariableDesignationSyntaxWrapper`

rather than `isSingleVariable` I suggest `isInsideParanthesizedVariableDesignation`, which would reverse the current boolean checks & values - WDYT?